src/server/game/Entities/Player/Player.cpp

```diff
@@ -2937,6 +2942,11 @@ bool Player::addTalent(uint32 spellId, uint8 addSpecMask, uint8 oldTalentRank)
         if (itr->second->State != PLAYERSPELL_NEW)
             itr->second->State = PLAYERSPELL_CHANGED;
 
+        if (GetActiveSpecMask() & addSpecMask)
+        {
+            m_usedTalentCount += (talentPos->rank + 1) - oldTalentRank;
+        }
```

Comment:
```suggestion
        if (GetActiveSpecMask() & addSpecMask)
            m_usedTalentCount += (talentPos->rank + 1) - oldTalentRank;
```

https://www.azerothcore.org/wiki/cpp-code-standards